build_tools/third_party/tensorflow/tensorflow/compiler/mlir/xla/CMakeLists.txt

```diff
@@ -21,6 +21,9 @@ set(TF_MLIR_XLA_COPTS_BASE
   "-I${CMAKE_BINARY_DIR}/build_tools/third_party/tensorflow/"
 )
 
+file(MAKE_DIRECTORY "${CMAKE_BINARY_DIR}/build_tools/third_party/tensorflow/tensorflow/compiler/mlir/xla/ir")
```

Comment:
Could you add a comment here explaining why this is needed? When I tried building again just now (after clearing my build cache), these `MAKE_DIRECTORY` lines didn't seem to be necessary. I could be missing something or might be building a different set of targets though.

Reply:
Building with `IREE_BUILD_COMPILER` set to `ON`, and further introducing a dependency on `iree-translate`, the build fails as document in issue #149. AFAIR all targets added via `external_tablegen_library` fail otherwise.

Reply:
Okay, I tried again after _really_ clearing my build cache and these lines do appear to be required:

```
code/iree$ rm -rf build/
code/iree$ cmake -B build/ -DIREE_BUILD_COMPILER=ON .
code/iree$ cmake --build build/ --target tensorflow_mlir_xla
```

fails without the `MAKE_DIRECTORY` lines, succeeds with all of the changes in this file.

Perhaps add a comment like
`# Create directories in the build tree for external_tablegen_library targets to use.`
just so future readers of this file can see what is happening. We could also move the `MAKE_DIRECTORY` step into `external_tablegen_library` if we end up using it in more places, but I think this is fine for now.

Reply:
Added the comment as requested. I further agree that we should move `MAKE_DIRECTORY` if there are additional targets relying on `external_tablegen_library` beside `tensorflow_mlir_xla`. Will do so if I trip over it.

Reply:
Thanks! I can merge this once the CLA is signed.